docs/build/mapping/store.md

```diff
@@ -0,0 +1,128 @@
+  remove(entity: string, id: string): Promise<void>;
+}
+```
+
+## Get Record
+
+`get(entity: string, id: string): Promise<Entity | null>;`
+
+This allows get an record of the entity with its `id`.
+
+```typescript
+const id = block.block.header.hash.toString();
+await store.get(`StarterEntity`, id);
+```
+
+## Get All Records by Field
+
+`getByField(entity: string, field: string, value: any): Promise<Entity[]>;`
+
+This returns all of the matching records for the specific entity that matches a given search.
+
+```typescript
+// Get all records with field1 == 50
+await store.getByField(`StarterEntity`, "field1", 50);
+```
+
+Please note, the third parameter also accept array, you can consider this similar like `bulkGet` with OR search.
+To get a list of records with `field1` equal to 50, 100 or 150
+
+```typescript
+// Get all records with field1 == 50 OR field1 == 100 OR field1 == 150
+await store.getByField("StarterEntity", "field1", [50, 100, 150]);
+```
+
+## Get First Record by Field
+
+`getOneByField(entity: string, field: string, value: any): Promise<Entity | null>;`
+
+This returns the first matching record for the specific entity that matches a given search.
+
+```typescript
+const field1Value = 50;
+await store.getOneByField(`StarterEntity`, `field1`, field1Value);
+```
+
+## Upsert (Create and Update) Record
+
+`set(entity: string, id: string, data: Entity): Promise<void>;`
+
+This allows user to create a single record, if the record already exist this will overwrite its record.
+
+```typescript
+const id = block.block.header.hash.toString();
+await store.set(`StarterEntity`,id, {field1: 50, ...})
+```
+
+## Bulk Create Records
+
+`bulkCreate(entity: string, data: Entity[]): Promise<void>;`
+
+This allows to create multiple records for specified entity, but it will not overwrite existing records.
+
+```typescript
+await store.bulkCreate(`StarterEntity`,[
+    {id: 1, field1: 50, ...},
+    {id: 2, field1: 100, ...},
+    {id: 3, field1: 150, ...}
+])
+```
+
+## Bulk Upsert (Create and Update) Records
+
+`bulkUpdate(entity: string, data: Entity[], fields?: string[]): Promise<void>;`
+
+This allows to update multiple records for specified entity, if will create the records if they are not exist.
+
+```typescript
+await store.bulkUpdate(`StarterEntity`,[
+    {id: 1, field1: 99, ...},
+    {id: 2, field1: 199, ...},
+    {id: 3, field1: 299, ...}
+])
+```
+
+The 3rd parameter is optional, allow user provide a list of fields they wish to be updated, and other fields will be ignored.
+
+For example, only `field5` will be updated.
+
+```typescript
+await store.bulkUpdate(`StarterEntity`,[
+    {id: 1, field1: 99, field5: true, ...},
+    {id: 2, field1: 199, field5: false,...},
+    ['field5']
+])
+```
+
+Please note, this fields feature currently not working with any automated historical indexing, it will overwrite all attributes. To disable automated historical indexing, please enable `--disable-historical=true` parameter on `subql-node`.
```

Comment:
```suggestion
Please note, this fields feature is not working currently with any automated historical indexing. It will overwrite all attributes. To disable automated historical indexing, please enable `--disable-historical=true` parameter on `subql-node`.
```